porch/controllers/remoterootsyncsets/pkg/remoteclient/getremoteclient.go

```diff
@@ -298,14 +366,57 @@ func (r *RemoteClientGetter) getGKERESTConfig(ctx context.Context, cluster *unst
 	restConfig.Host = "https://" + endpoint
 	klog.Infof("Host endpoint is %s", restConfig.Host)
 
-	tokenSource, err := r.getConfigConnectorContextTokenSource(ctx, cluster.GetNamespace())
+	tokenSource, err := r.getConfigConnectorTokenSource(ctx, cluster.GetNamespace())
+	if err != nil {
+		return nil, fmt.Errorf("error building authentication token provider: %w", err)
+	}
+	token, err := tokenSource.Token()
+	if err != nil {
+		return nil, fmt.Errorf("error getting authentication token: %w", err)
+	}
+	restConfig.BearerToken = token.AccessToken
+
+	return restConfig, nil
+}
+
+// getHubMembershipRESTConfig builds a rest.Config for accessing the specified cluster through connect gateway.
+func (r *RemoteClientGetter) getHubMembershipRESTConfig(ctx context.Context, cluster *unstructured.Unstructured) (*rest.Config, error) {
+	restConfig := &rest.Config{}
+
+	// TODO: We could really use a selfLink field here!
+
+	projectID := cluster.GetAnnotations()["cnrm.cloud.google.com/project-id"]
+	if projectID == "" {
+		return nil, fmt.Errorf("cannot determine project-id for object")
+	}
+
+	membershipName, _, err := unstructured.NestedString(cluster.Object, "spec", "resourceID")
+	if err != nil {
+		return nil, fmt.Errorf("failed to get spec.resourceID: %w", err)
+	}
+	if membershipName == "" {
+		return nil, fmt.Errorf("spec.resourceID field was not set")
+	}
+
+	tokenSource, err := r.getConfigConnectorTokenSource(ctx, cluster.GetNamespace())
 	if err != nil {
 		return nil, fmt.Errorf("error building authentication token provider: %w", err)
 	}
+
+	projectInfo, err := r.projectCache.LookupByProjectID(ctx, projectID, tokenSource)
+	if err != nil {
+		return nil, err
+	}
+
+	host := fmt.Sprintf("https://connectgateway.googleapis.com/v1/projects/%d/locations/global/memberships/%s", projectInfo.ProjectNumber, membershipName)
```

Comment:
I only see membershipName as part of the `host` here.  Does membership maps 1:1 to a cluster that is part of Hub ?
If that's the question, I have a follow up question: How do I discover all the memberships ?

I might block some time today for us to understand this machinery.

/cc @ChristopherFry

Reply:
I guess, hitting the GCP API using the workload identity of the `config-controller` would one way to do this I guess.